### PR TITLE
a few additional units

### PR DESCRIPTION
**Pull request by Alex-Jordan**
In upgrading my production server, I had to take notice of all the customizations we've added. These are some that are worth contributing. I was especially surprised to see that gallons are missing.

### CODE REVIEW

File: lib/Units.pm [hunk 1]
```diff
@@ -180,6 +181,11 @@ our %known_units = (
 		factor => 1E-15,
 		m      => 1
 	},
+	meter => {
+		factor  => 1,
+		m       => 1,
+		aliases => ['meters']
+	},
```
Comment (drgrice1) [inline lib/Units.pm:188]: This should not be added as a new unit.  Instead "meter" and "meters" should be added directly as aliases of "m" (defined on line 42 of this file).

**Comment by Alex-Jordan** [comment]
Oh I didn't realize the fundamental units had this style of definition too.

**Comment by drgrice1** [comment]
Yeah, all of the fundamental units are defined first in the`%known_units` hash.

**Review by pstaabp — APPROVED** [review]

File: lib/Units.pm [hunk 2]
```diff
@@ -233,14 +235,19 @@ our %known_units = (
 		m       => 3,
 		aliases => ['cups']
 	},
+	gallon => {
+		factor  => 0.00378541,
+		m       => 3,
+		aliases => [ 'gallons', 'gal' ]
+	},
```
Comment (drgrice1) [inline lib/Units.pm:242]: I think there are some other related units that could be added here.  Pint and quart go with these volume measurements, and are equally standard with cup and gallon.  There are also teaspoon and tablespoon, that are rather common.  If you really want to go the distance there are also pottle, barrel, shot, gill, hogshead, etc.  I haven't heard of some of those before.  Some fun names too!  See [customary US units](https://en.wikipedia.org/wiki/United_States_customary_units).

**Review by drgrice1 — APPROVED** [review]
Looks good.  But how far do we go with this?

File: lib/Units.pm (hunk 2 above)
Comment (pstaabp) [reply lib/Units.pm:242]: I'll have a hogsheads of mead with lunch today. 😄

**Comment by Alex-Jordan** [comment]
I'm fine with going further. But here are some possible principles that
could restrain us.

Any reasonably common unit for stem problems should be in there. Gallons
counts, hogshead not.

For any unit in there, all reasonable aliases in the context of what a
student might type should be available. Like "meter".

Individual problems can always add new units like hogshead.

Questions I have though:
* What about metre and litre?
* We have L and ml but not mL.
* Students can want to type periods after some units, should we cater to
that?

As far as this pr goes these are currently just the units we've added at
PCC.

**Comment by pstaabp** [comment]
I'm prepping a PR to this branch for pints and quarts and adding some tests for these.  Should I add 'litre(s)', 'metre(s)' etc. to this as well?

**Comment by pstaabp** [comment]
@Alex-Jordan I put a PR into your branch.  Also added the new units to the help file.

File: lib/Units.pm (hunk 2 above)
Comment (duffee) [reply lib/Units.pm:242]: For those that like extreme edge cases, the Gal is the acceleration of 1 cm/s^2, but was deprecated by the Int. Ast. Union in 1989.  Probably an idea to include a comment that the gallon is US, not Imperial.

https://en.wikipedia.org/wiki/Gal_(unit)
https://www.iau.org/publications/proceedings_rules/units/

**Comment by Alex-Jordan** [comment]
I accidentally rebased on 2.18 instead of merging 2.18 here. The GitHub diff looks weird now. I'll see if I can fix it.

**Comment by Alex-Jordan** [comment]
I think I straightened this out.

**Comment by drgrice1** [comment]
A rebase should work fine.

**Comment by Alex-Jordan** [comment]
It works better when you see there was a conflict and fix it, instead of just pushing it half-rebased.

File: lib/Units.pm (hunk 2 above)
Comment (Alex-Jordan) [reply lib/Units.pm:253]: @duffee In the help file, I put "US " in front of the title for each of gallon, quart, pint, and cup.

**Pull request merged**